src/api/routes/transcripts.ts

```diff
@@ -114,22 +114,27 @@ export async function getSummariesAndNameMap(transcriptId: string): Promise<{
     where: { handlebarName: [...handlebarsSet] },
     include: [{
       model: db.User,
-      attributes: ['name'],
+      attributes: ['name','id'],
       where: { name: { [Op.ne]: null } } // Ensure user's name is not null
     }],
   });
 
-  let nameMap: TranscriptNameMap = tnm;
-  
+  let nameMap: TranscriptNameMap = [];
   // check if the handlebar names are presented in the returned array of transcriptNameMap
   // if not, create an object of this handlebar with null user, 
   // otherwise when Handlebars.js compile it will return empty
   for (const handlebar of [...handlebarsSet]) {
-    if (![...nameMap].some(nm => nm.handlebarName === handlebar)) {
+    const entries = tnm.filter(e => e.handlebarName === handlebar);
+    if (entries.length) {
+      const mappedEntries = entries.map(e => ({
```

Comment:
don't use `map`. do

```
invariant(entries.length == 1);
manMap.push( ...entries[0]... );
```

Reply:
![image](https://github.com/yuanjian-org/app/assets/83266861/47808923-203a-4ce3-851e-e36feede4718)
![image](https://github.com/yuanjian-org/app/assets/83266861/f5b7e498-f301-4509-98c3-0ddb5a6011f6)

the returned array of objects is shown above. So an object with specifically assigned values was pushed instead a spread operator